drgn_tools/dm.py

```diff
@@ -104,6 +106,70 @@ def show_dm(prog: Program) -> None:
     print_table(output)
 
 
+def dm_table(dm: Object) -> Object:
+    """
+    return the ``struct dm_table *``
+
+    There were two definitions of ``struct dm_table`` before commit
+    1d3aa6f683b1("dm: remove dummy definition of 'struct dm_table'")
+    which was included in v4.10, specify file for the correct symbol.
+    """
+    if kernel_version(dm.prog_) < (4, 10, 0):
+        table_type = dm.prog_.type(
+            "struct dm_table *", "drivers/md/dm-table.c"
+        )
```

Comment:
Interestingly enough, I took a look at the UEK4 CTF archives, and found that the "undefined" variant of dm_table is not even included in the CTF archive, since it never gets used by anything. I guess CTF is "smart enough" to handle this situation.

Also, UEK4 is not currently planned to be supported for CTF, so that's not much of an issue anyway.

Reply:
CTF is awesome!